File: Olympiad/COCI/official/2009/contest4_solutions/datum.pas
program datum;

const dana_u_mjesecu : array[1..12] of integer = (31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31);
const imena : array[0..6] of string = ('Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday');

var
   dd, mm, dan, m : longint;

begin
   read(dd, mm);
   dan := 2+dd;
   for m:=1 to mm-1 do
      dan := dan + dana_u_mjesecu[m];
   writeln(imena[dan mod 7]);
end.

File: Olympiad/COCI/official/2009/contest4_solutions/datum.cpp
#include <cstdio>

int dana_u_mjesecu[13] = { 0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
char *imena[] = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday" };

int main( void ) {
   int dd, mm;
   scanf( "%d%d", &dd, &mm );
   
   int dan = 2 + dd;
   for( int m = 1; m < mm; ++m ) dan += dana_u_mjesecu[m];

   printf( "%s\n", imena[dan%7] );

   return 0;
}
